BackEnd/Controllers/userController.js

```diff
@@ -1,25 +1,130 @@
 const User = require("../models/userModel");
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
 
 exports.signUp = async (req, res) => {
   await User.create(req.body);
   res.status(201).json({
     success: true,
-    // data: newUser,
     message: "User created successfully",
   });
 };
 
 exports.logIn = async (req, res) => {
-  const user = await User.findOne({
-    email: req.body.email,
-    password: req.body.password,
-  });
+  const { email, password } = req.body;
+  try {
+    // Find the user by email
+    const user = await User.findOne({ email }).lean();
+    if (!user) {
+      return res.status(401).json({ error: "Authentication failed" });
+    }
+
+    // Compare the provided password with the stored hashed password
+    const passwordMatch = await bcrypt.compare(password, user.password);
+    if (!passwordMatch) {
+      return res.status(401).json({ error: "Authentication failed" });
+    }
+
+    // Exclude the password field from the user object
+    const { password: _, ...others } = user;
+
+    const token = jwt.sign({ data: others }, process.env.JWT_SECRET_KEY, {
+      expiresIn: process.env.EXPIRES_IN,
+    });
+
+    res.status(200).json({ access_token: token, others });
+  } catch (error) {
+    res.status(400).json({ error: "Login failed" });
+  }
+};
+
+exports.verifyToken = function (req, res, next) {
```

Comment:
create a new folder named Middleware and then create new file called authMiddleware.js. Then add this function to authMiddleware file.